stdlib/Random/test/runtests.jl

```diff
@@ -1250,6 +1250,12 @@ end
     @test length(xs) == 3
 end
 
+@testset "float32 bug" begin
+    a = rand(Float32, 4)
+    @test a[1] != a[2]
+    @test a[3] != a[4]
```

Comment:
```julia
    @test length(unique(rand(Float32, 14))) > 10
```
would be less specific and should have a reasonably low probability of accidental failure as well. (Could be adjusted by modifying the 10, of course.)

Reply:
Yeah, that seems to be less likely to hit a random failure:
```
julia> all(length(unique(rand(Float32, 14))) > 10 for _ in 1:100_000_000)
true

julia> all(allunique(rand(Float32, 4)) for _ in 1:100_000_000)
false
```